mobile/cmp/popover/Popover.js

```diff
@@ -0,0 +1,196 @@
+ */
+import {hoistCmp, HoistModel, useLocalModel} from '@xh/hoist/core';
+import {observable, action, makeObservable} from '@xh/hoist/mobx';
+import {div} from '@xh/hoist/cmp/layout';
+import {throwIf} from '@xh/hoist/utils/js';
+import {createObservableRef} from '@xh/hoist/utils/react';
+import {isNil, isFunction} from 'lodash';
+import {isValidElement} from 'react';
+import {usePopper} from 'react-popper';
+import ReactDom from 'react-dom';
+import classNames from 'classnames';
+import PT from 'prop-types';
+
+import './Popover.scss';
+
+/**
+ * Popovers display floating content next to a target element.
+ *
+ * The API is based on a stripped-down version of Blueprint's Popover component
+ * that is used on Desktop. Popover is built on top of the Popper.js library.
+ *
+ * @see https://popper.js.org/
+ */
+export const [Popover, popover] = hoistCmp.withFactory({
+    displayName: 'Popover',
+    model: false,
+    className: 'xh-popover',
+
+    render({
+        className,
+        target,
+        content,
+        isOpen,
+        onInteraction,
+        withMask = false,
+        position = 'auto',
+        popperOptions
+    }) {
+        throwIf(!isValidElement(target), 'Popover requires a `target` component');
+        throwIf(!isValidElement(content), 'Popover requires a `content` component');
+
+        const impl = useLocalModel(() => new LocalModel()),
+            popper = usePopper(impl.targetEl, impl.contentEl, {
+                placement: impl.menuPositionToPlacement(position),
+                strategy: 'fixed',
+                modifiers: [
+                    {name: 'preventOverflow', options: {padding: 10}}
+                ],
+                ...popperOptions
+            });
+
+        // Respect controlled mode prop
+        if (!isNil(isOpen)) {
+            impl.setControlledMode(isOpen, onInteraction);
+        }
+
+        return div({
+            className,
+            items: [
+                div({
+                    ref: impl.targetRef,
+                    className: 'xh-popover__target-wrapper',
+                    item: target,
+                    onClick: () => impl.toggleOpen()
+                }),
+                div({
+                    ref: impl.contentRef,
+                    omit: !impl.isOpen,
+                    style: popper?.styles?.popper,
+                    className: 'xh-popover__content-wrapper',
+                    items: content
+                }),
+                div({
+                    omit: !impl.isOpen,
+                    className: classNames(
+                        'xh-popover__content-overlay',
+                        withMask ? 'xh-popover__content-overlay--mask' : null
+                    ),
+                    onClick: () => impl.setIsOpen(false)
+                })
+            ]
+        });
+    }
+});
+
+Popover.propTypes = {
+    /** Component to display inside the popover */
+    content: PT.element,
```

Comment:
Wondering if this should be something that could be parsed as  `elementFromContent`.   We use that in a number of places in the API.  Could use the same for target.

We could also expand `elementFromContent` to take an element itself as well as a Component/function.  Not sure why we never supported that.  Would be useful in other contexts as well.

Reply:
Agree strongly (and see Tom has already made the change) that `content` should accept the same type of things across its usages in our container components.

And the fact that this is the _component_, as opposed to the tab or navigator models, makes me also agree that our util should support an element itself, so you can use it as you would `items`.  Thanks for making that change, too.

The one concern is that we deliberately had `content` taking a function that returns an element in the model case to support lazy rendering, as you are often configuring/constructing your model well prior to any content actually being rendered (if it ever is).  It seems that we should emphasize that usage so people don't start configuring models with elements and defeat the built-in laziness.  (I hope I am understanding / describing this correctly.)

Reply:
i.e. emphasize (as succinctly as possible) the reasons for wrapping in a closure - when and why you would do that vs. when you would provide an element directly.

Did we think about just going with `items` for the content?  Or is the relationship between `target` and `content` such that neither has a clearer claim to being the child than the other?

Reply:
Maybe I'm misunderstanding, but I thought that even if you provide an element as content for e.g. TabContainer, it won't get rendered until it becomes the active tab? Or, to put it another way, its not rendered until it actually gets added to the DOM in accordance with its RenderMode?

Reply:
Tom -- w.r.t. to your question, I think you have it right --  the element generation should be pretty lightweight.  

According to my understanding, no actual Component rendering, render methods etc. are going to happen until React actually tries to mount that element.  So things would still be quite lazy.   It's really just a matter of if you are ready to execute the code itself to specify the element (vs. doing that a bit later).

So really 
      
`content: () => panel({})`
 `content: panel({})`.  

are not that different.  It seems like it would be hard to get yourself in too much trouble specifying an actual element declaratively.

Reply:
regarding ATM's question about `items` vs. `target` -- it seems like we are trying to mirror the BP desktop API, for better or worse.   If we weren't doing that, I think we might go in a number of different directions.

Interesting to our discussion about returning elements that the new blueprint api actually *requires* a `renderTarget` (vs. target)   Weird that the always rendered portion is so declaratively lazy, and not the content (which may never be displayed).